recipes/lora_finetune_single_device.py

```diff
@@ -514,11 +518,20 @@ def save_checkpoint(self, epoch: int) -> None:
             "peft_type": "LORA",
         }
         ckpt_dict.update({utils.ADAPTER_CONFIG: adapter_config})
+
         self._checkpointer.save_checkpoint(
             ckpt_dict,
             epoch=epoch,
-            intermediate_checkpoint=(epoch + 1 < self.total_epochs),
+            intermediate_checkpoint=is_intermediate_epoch,
+            adapter_only=self._save_adapter_weights_only,
         )
+        if not is_intermediate_epoch:
+            log.info(
+                "Saving final model checkpoint."
+                "Please note that you have set save_adapter_weights_only=True, so only adapter weights will be saved."
+                "You need to merge the adapter weights into your base model for further use. "
+                f"See {type(self._checkpointer).__name__}"
+            )
```

Comment:
Good catch. Sorry, my original suggestion missed this.

Do we still think it's a good idea to move this into the checkpointer, rather than recipes?

Reply:
@SalmanMohammadi 

Indeed, rather than writing similar code each time we create a new recipe, it might be better to create it with a checkpointer. What do you think?

Reply:
> Indeed, rather than writing similar code each time we create a new recipe, it might be better to create it with a checkpointer. What do you think?

Good point, I'm inclined to agree with this (kinda similar to my other comment, it's nice to keep the recipe files themselves as clean as possible)